### ISSUE TITLE
touchpad

### ISSUE THREAD

**Issue by Manuelibt** [sarmbruster/thinkpad_x1_yoga_rotation#7]
hi,
i have seen that you stopped to develope that script, but maybe you can still help me. 
I am on ubuntu 17.04 and i am using gnome as you are. 
It is working for me nearly as well as with your script on ubuntu with kde, except for two trifles: 

1. Touchscreen is not detecting double clicks e.g. in nautilus
2. touchpad is still active in tabletmode

How are you handling this? Do you have the same issues on ubuntu 17.10?

Thank you in advance.

**Comment by hensur** [comment]
Maybe I can help you with the TouchPad issue if you still need help.
Can you post the output of:
xinput --list --name-only

I don't own a Thinkpad X1 Yoga, but on my Thinkpad Yoga 370 the TouchPad had a different name and was not detected by the script. The command above could clarify this.

**Comment by sarmbruster** [comment]
Meanwhile I'm on Ubuntu 17.10 using Wayland. Wayland seems not to support lots of the tools I was originally using in the script. `xinput --list --name-only` returns:
```Virtual core pointer
Virtual core XTEST pointer
xwayland-pointer:13
xwayland-relative-pointer:13
xwayland-touch:13
xwayland-stylus:13
xwayland-eraser:13
xwayland-cursor:13
Virtual core keyboard
Virtual core XTEST keyboard
xwayland-keyboard:13```
Which is not super helpful....

**Comment by hensur** [comment]
Yeah, I'm still using X because I had some issues with Xournal and Wayland.
I don't know if there is a tool like xinput for Wayland. A tool like this would probaly have to talk to the Compositor directly, since the Desktop Environment is now responsible for the Input Devices and will control them directly using libinput. There is no layer between the DE and the devices anymore, which once was the X Server. So a script like this would have to somehow tell Gnome to disable the Touchpad directly.
I think I will continue using X while there is no real way of controlling these devices on Wayland.
